// xendit-node/payment_method/models/OverTheCounterChannelCode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OverTheCounterChannelCodeToJSON = exports.OverTheCounterChannelCodeFromJSONTyped = exports.OverTheCounterChannelCodeFromJSON = exports.OverTheCounterChannelCode = void 0;
/**
 * Over The Counter Channel Code
 * @export
 */
exports.OverTheCounterChannelCode = {
    _7Eleven: '7ELEVEN',
    _7ElevenCliqq: '7ELEVEN_CLIQQ',
    Cebuana: 'CEBUANA',
    Ecpay: 'ECPAY',
    Palawan: 'PALAWAN',
    Mlhuillier: 'MLHUILLIER',
    EcpayDragonloan: 'ECPAY_DRAGONLOAN',
    Lbc: 'LBC',
    EcpaySchool: 'ECPAY_SCHOOL',
    RdPawnshop: 'RD_PAWNSHOP',
    Cvm: 'CVM',
    Ussc: 'USSC',
    SmBills: 'SM_BILLS',
    RobinsonsBills: 'ROBINSONS_BILLS',
    Alfamart: 'ALFAMART',
    Indomaret: 'INDOMARET'
};
function OverTheCounterChannelCodeFromJSON(json) {
    return OverTheCounterChannelCodeFromJSONTyped(json, false);
}
exports.OverTheCounterChannelCodeFromJSON = OverTheCounterChannelCodeFromJSON;
function OverTheCounterChannelCodeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.OverTheCounterChannelCodeFromJSONTyped = OverTheCounterChannelCodeFromJSONTyped;
function OverTheCounterChannelCodeToJSON(value) {
    return value;
}
exports.OverTheCounterChannelCodeToJSON = OverTheCounterChannelCodeToJSON;
