// xendit-node/payment_method/models/PaymentChannelListLinksInnerAllOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentChannelListLinksInnerAllOfToJSON = exports.PaymentChannelListLinksInnerAllOfFromJSONTyped = exports.PaymentChannelListLinksInnerAllOfFromJSON = exports.instanceOfPaymentChannelListLinksInnerAllOf = exports.PaymentChannelListLinksInnerAllOfRelEnum = void 0;
var runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.PaymentChannelListLinksInnerAllOfRelEnum = {
    First: 'first',
    Last: 'last',
    Next: 'next',
    Prev: 'prev'
};
/**
 * Check if a given object implements the PaymentChannelListLinksInnerAllOf interface.
 */
function instanceOfPaymentChannelListLinksInnerAllOf(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfPaymentChannelListLinksInnerAllOf = instanceOfPaymentChannelListLinksInnerAllOf;
function PaymentChannelListLinksInnerAllOfFromJSON(json) {
    return PaymentChannelListLinksInnerAllOfFromJSONTyped(json, false);
}
exports.PaymentChannelListLinksInnerAllOfFromJSON = PaymentChannelListLinksInnerAllOfFromJSON;
function PaymentChannelListLinksInnerAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': !(0, runtime_1.exists)(json, 'href') ? undefined : json['href'],
        'rel': !(0, runtime_1.exists)(json, 'rel') ? undefined : json['rel'],
        'method': !(0, runtime_1.exists)(json, 'method') ? undefined : json['method'],
    };
}
exports.PaymentChannelListLinksInnerAllOfFromJSONTyped = PaymentChannelListLinksInnerAllOfFromJSONTyped;
function PaymentChannelListLinksInnerAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'href': value.href,
        'rel': value.rel,
        'method': value.method,
    };
}
exports.PaymentChannelListLinksInnerAllOfToJSON = PaymentChannelListLinksInnerAllOfToJSON;
