// xendit-node/payment_method/models/PaymentChannelListLinksInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentChannelListLinksInnerToJSON = exports.PaymentChannelListLinksInnerFromJSONTyped = exports.PaymentChannelListLinksInnerFromJSON = exports.instanceOfPaymentChannelListLinksInner = exports.PaymentChannelListLinksInnerRelEnum = void 0;
var runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.PaymentChannelListLinksInnerRelEnum = {
    First: 'first',
    Last: 'last',
    Next: 'next',
    Prev: 'prev'
};
/**
 * Check if a given object implements the PaymentChannelListLinksInner interface.
 */
function instanceOfPaymentChannelListLinksInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfPaymentChannelListLinksInner = instanceOfPaymentChannelListLinksInner;
function PaymentChannelListLinksInnerFromJSON(json) {
    return PaymentChannelListLinksInnerFromJSONTyped(json, false);
}
exports.PaymentChannelListLinksInnerFromJSON = PaymentChannelListLinksInnerFromJSON;
function PaymentChannelListLinksInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': !(0, runtime_1.exists)(json, 'href') ? undefined : json['href'],
        'rel': !(0, runtime_1.exists)(json, 'rel') ? undefined : json['rel'],
        'method': !(0, runtime_1.exists)(json, 'method') ? undefined : json['method'],
    };
}
exports.PaymentChannelListLinksInnerFromJSONTyped = PaymentChannelListLinksInnerFromJSONTyped;
function PaymentChannelListLinksInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'href': value.href,
        'rel': value.rel,
        'method': value.method,
    };
}
exports.PaymentChannelListLinksInnerToJSON = PaymentChannelListLinksInnerToJSON;
