// xendit-node/payout/apis/Payout.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutApi = void 0;
var runtime = require("../../runtime");
var models_1 = require("../models");
/**
 *
 */
var PayoutApi = /** @class */ (function (_super) {
    __extends(PayoutApi, _super);
    function PayoutApi(_a) {
        var secretKey = _a.secretKey, xenditURL = _a.xenditURL;
        var _this = this;
        var conf = new runtime.Configuration({
            basePath: xenditURL || 'https://api.xendit.co'
        });
        _this = _super.call(this, conf) || this;
        _this.secretKey = "";
        _this.xenditURL = "";
        _this.secretKey = secretKey;
        return _this;
    }
    /**
     * API to cancel requested payouts that have not yet been sent to partner banks and e-wallets. Cancellation is possible if the payout has not been sent out via our partner and when payout status is ACCEPTED.
     */
    PayoutApi.prototype.cancelPayoutRaw = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.id === null || requestParameters.id === undefined) {
                            throw new runtime.RequiredError('id', 'Required parameter requestParameters.id was null or undefined when calling cancelPayout.');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters["Authorization"] = "Basic " + btoa(this.secretKey + ":");
                        if (requestParameters.idempotencyKey !== undefined && requestParameters.idempotencyKey !== null) {
                            headerParameters['idempotency-key'] = String(requestParameters.idempotencyKey);
                        }
                        return [4 /*yield*/, this.request({
                                path: "/v2/payouts/{id}/cancel".replace("{".concat("id", "}"), encodeURIComponent(String(requestParameters.id))),
                                method: 'POST',
                                headers: headerParameters,
                                query: queryParameters,
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, models_1.GetPayouts200ResponseDataInnerFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * API to cancel requested payouts that have not yet been sent to partner banks and e-wallets. Cancellation is possible if the payout has not been sent out via our partner and when payout status is ACCEPTED.
     */
    PayoutApi.prototype.cancelPayout = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.cancelPayoutRaw(requestParameters)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * API to send money at scale to bank accounts & eWallets
     */
    PayoutApi.prototype.createPayoutRaw = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.idempotencyKey === null || requestParameters.idempotencyKey === undefined) {
                            throw new runtime.RequiredError('idempotencyKey', 'Required parameter requestParameters.idempotencyKey was null or undefined when calling createPayout.');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters["Authorization"] = "Basic " + btoa(this.secretKey + ":");
                        headerParameters['Content-Type'] = 'application/json';
                        if (requestParameters.idempotencyKey !== undefined && requestParameters.idempotencyKey !== null) {
                            headerParameters['idempotency-key'] = String(requestParameters.idempotencyKey);
                        }
                        if (requestParameters.forUserId !== undefined && requestParameters.forUserId !== null) {
                            headerParameters['for-user-id'] = String(requestParameters.forUserId);
                        }
                        return [4 /*yield*/, this.request({
                                path: "/v2/payouts",
                                method: 'POST',
                                headers: headerParameters,
                                query: queryParameters,
                                body: (0, models_1.CreatePayoutRequestToJSON)(requestParameters.data),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, models_1.GetPayouts200ResponseDataInnerFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * API to send money at scale to bank accounts & eWallets
     */
    PayoutApi.prototype.createPayout = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.createPayoutRaw(requestParameters)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * API to fetch the current status, or details of the payout
     */
    PayoutApi.prototype.getPayoutByIdRaw = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.id === null || requestParameters.id === undefined) {
                            throw new runtime.RequiredError('id', 'Required parameter requestParameters.id was null or undefined when calling getPayoutById.');
                        }
                        queryParameters = {};
                        headerParameters = {};
                        headerParameters["Authorization"] = "Basic " + btoa(this.secretKey + ":");
                        if (requestParameters.idempotencyKey !== undefined && requestParameters.idempotencyKey !== null) {
                            headerParameters['idempotency-key'] = String(requestParameters.idempotencyKey);
                        }
                        return [4 /*yield*/, this.request({
                                path: "/v2/payouts/{id}".replace("{".concat("id", "}"), encodeURIComponent(String(requestParameters.id))),
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, models_1.GetPayouts200ResponseDataInnerFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * API to fetch the current status, or details of the payout
     */
    PayoutApi.prototype.getPayoutById = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPayoutByIdRaw(requestParameters)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * API providing the current list of banks and e-wallets we support for payouts for both regions
     */
    PayoutApi.prototype.getPayoutChannelsRaw = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        queryParameters = {};
                        if (requestParameters.currency !== undefined) {
                            queryParameters['currency'] = requestParameters.currency;
                        }
                        if (requestParameters.channelCategory) {
                            queryParameters['channel_category'] = requestParameters.channelCategory.join(runtime.COLLECTION_FORMATS["csv"]);
                        }
                        if (requestParameters.channelCode !== undefined) {
                            queryParameters['channel_code'] = requestParameters.channelCode;
                        }
                        headerParameters = {};
                        headerParameters["Authorization"] = "Basic " + btoa(this.secretKey + ":");
                        if (requestParameters.idempotencyKey !== undefined && requestParameters.idempotencyKey !== null) {
                            headerParameters['idempotency-key'] = String(requestParameters.idempotencyKey);
                        }
                        return [4 /*yield*/, this.request({
                                path: "/payouts_channels",
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return jsonValue.map(models_1.ChannelFromJSON); })];
                }
            });
        });
    };
    /**
     * API providing the current list of banks and e-wallets we support for payouts for both regions
     */
    PayoutApi.prototype.getPayoutChannels = function (requestParameters) {
        if (requestParameters === void 0) { requestParameters = {}; }
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPayoutChannelsRaw(requestParameters)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    /**
     * API to retrieve all matching payouts with reference ID
     */
    PayoutApi.prototype.getPayoutsRaw = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var queryParameters, headerParameters, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        if (requestParameters.referenceId === null || requestParameters.referenceId === undefined) {
                            throw new runtime.RequiredError('referenceId', 'Required parameter requestParameters.referenceId was null or undefined when calling getPayouts.');
                        }
                        queryParameters = {};
                        if (requestParameters.referenceId !== undefined) {
                            queryParameters['reference_id'] = requestParameters.referenceId;
                        }
                        if (requestParameters.limit !== undefined) {
                            queryParameters['limit'] = requestParameters.limit;
                        }
                        if (requestParameters.afterId !== undefined) {
                            queryParameters['after_id'] = requestParameters.afterId;
                        }
                        if (requestParameters.beforeId !== undefined) {
                            queryParameters['before_id'] = requestParameters.beforeId;
                        }
                        headerParameters = {};
                        headerParameters["Authorization"] = "Basic " + btoa(this.secretKey + ":");
                        if (requestParameters.idempotencyKey !== undefined && requestParameters.idempotencyKey !== null) {
                            headerParameters['idempotency-key'] = String(requestParameters.idempotencyKey);
                        }
                        return [4 /*yield*/, this.request({
                                path: "/v2/payouts",
                                method: 'GET',
                                headers: headerParameters,
                                query: queryParameters,
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, new runtime.JSONApiResponse(response, function (jsonValue) { return (0, models_1.GetPayouts200ResponseFromJSON)(jsonValue); })];
                }
            });
        });
    };
    /**
     * API to retrieve all matching payouts with reference ID
     */
    PayoutApi.prototype.getPayouts = function (requestParameters) {
        return __awaiter(this, void 0, void 0, function () {
            var response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getPayoutsRaw(requestParameters)];
                    case 1:
                        response = _a.sent();
                        return [4 /*yield*/, response.value()];
                    case 2: return [2 /*return*/, _a.sent()];
                }
            });
        });
    };
    return PayoutApi;
}(runtime.BaseAPI));
exports.PayoutApi = PayoutApi;
