// xendit-node/payment_request/models/PaymentMethodReusability.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodReusabilityToJSON = exports.PaymentMethodReusabilityFromJSONTyped = exports.PaymentMethodReusabilityFromJSON = exports.PaymentMethodReusability = void 0;
/**
 *
 * @export
 */
exports.PaymentMethodReusability = {
    MultipleUse: 'MULTIPLE_USE',
    OneTimeUse: 'ONE_TIME_USE'
};
function PaymentMethodReusabilityFromJSON(json) {
    return PaymentMethodReusabilityFromJSONTyped(json, false);
}
exports.PaymentMethodReusabilityFromJSON = PaymentMethodReusabilityFromJSON;
function PaymentMethodReusabilityFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PaymentMethodReusabilityFromJSONTyped = PaymentMethodReusabilityFromJSONTyped;
function PaymentMethodReusabilityToJSON(value) {
    return value;
}
exports.PaymentMethodReusabilityToJSON = PaymentMethodReusabilityToJSON;
