// xendit-node/payment_request/models/PaymentRequestCaptureMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequestCaptureMethodToJSON = exports.PaymentRequestCaptureMethodFromJSONTyped = exports.PaymentRequestCaptureMethodFromJSON = exports.PaymentRequestCaptureMethod = void 0;
/**
 *
 * @export
 */
exports.PaymentRequestCaptureMethod = {
    Automatic: 'AUTOMATIC',
    Manual: 'MANUAL'
};
function PaymentRequestCaptureMethodFromJSON(json) {
    return PaymentRequestCaptureMethodFromJSONTyped(json, false);
}
exports.PaymentRequestCaptureMethodFromJSON = PaymentRequestCaptureMethodFromJSON;
function PaymentRequestCaptureMethodFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PaymentRequestCaptureMethodFromJSONTyped = PaymentRequestCaptureMethodFromJSONTyped;
function PaymentRequestCaptureMethodToJSON(value) {
    return value;
}
exports.PaymentRequestCaptureMethodToJSON = PaymentRequestCaptureMethodToJSON;
