// xendit-node/payment_request/models/PaymentRequestCountry.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequestCountryToJSON = exports.PaymentRequestCountryFromJSONTyped = exports.PaymentRequestCountryFromJSON = exports.PaymentRequestCountry = void 0;
/**
 *
 * @export
 */
exports.PaymentRequestCountry = {
    Id: 'ID',
    Ph: 'PH',
    Vn: 'VN',
    Th: 'TH',
    My: 'MY'
};
function PaymentRequestCountryFromJSON(json) {
    return PaymentRequestCountryFromJSONTyped(json, false);
}
exports.PaymentRequestCountryFromJSON = PaymentRequestCountryFromJSON;
function PaymentRequestCountryFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PaymentRequestCountryFromJSONTyped = PaymentRequestCountryFromJSONTyped;
function PaymentRequestCountryToJSON(value) {
    return value;
}
exports.PaymentRequestCountryToJSON = PaymentRequestCountryToJSON;
