// xendit-node/invoice/models/InvoiceStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceStatusToJSON = exports.InvoiceStatusFromJSONTyped = exports.InvoiceStatusFromJSON = exports.InvoiceStatus = void 0;
/**
 * Representing the status of an invoice.
 * @export
 */
exports.InvoiceStatus = {
    Pending: 'PENDING',
    Paid: 'PAID',
    Settled: 'SETTLED',
    Expired: 'EXPIRED'
};
function InvoiceStatusFromJSON(json) {
    return InvoiceStatusFromJSONTyped(json, false);
}
exports.InvoiceStatusFromJSON = InvoiceStatusFromJSON;
function InvoiceStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.InvoiceStatusFromJSONTyped = InvoiceStatusFromJSONTyped;
function InvoiceStatusToJSON(value) {
    return value;
}
exports.InvoiceStatusToJSON = InvoiceStatusToJSON;
