// xendit-node/invoice/models/InvoiceClientType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceClientTypeToJSON = exports.InvoiceClientTypeFromJSONTyped = exports.InvoiceClientTypeFromJSON = exports.InvoiceClientType = void 0;
/**
 * Representing the client type or source of an invoice.
 * @export
 */
exports.InvoiceClientType = {
    Dashboard: 'DASHBOARD',
    ApiGateway: 'API_GATEWAY',
    Integration: 'INTEGRATION',
    OnDemand: 'ON_DEMAND',
    Recurring: 'RECURRING',
    Mobile: 'MOBILE'
};
function InvoiceClientTypeFromJSON(json) {
    return InvoiceClientTypeFromJSONTyped(json, false);
}
exports.InvoiceClientTypeFromJSON = InvoiceClientTypeFromJSON;
function InvoiceClientTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.InvoiceClientTypeFromJSONTyped = InvoiceClientTypeFromJSONTyped;
function InvoiceClientTypeToJSON(value) {
    return value;
}
exports.InvoiceClientTypeToJSON = InvoiceClientTypeToJSON;
