// xendit-node/invoice/models/InvoicePaymentMethod.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoicePaymentMethodToJSON = exports.InvoicePaymentMethodFromJSONTyped = exports.InvoicePaymentMethodFromJSON = exports.InvoicePaymentMethod = void 0;
/**
 * Representing the payment method used for an invoice.
 * @export
 */
exports.InvoicePaymentMethod = {
    Pool: 'POOL',
    CallbackVirtualAccount: 'CALLBACK_VIRTUAL_ACCOUNT',
    CreditCard: 'CREDIT_CARD',
    RetailOutlet: 'RETAIL_OUTLET',
    QrCode: 'QR_CODE',
    Qris: 'QRIS',
    Ewallet: 'EWALLET',
    DirectDebit: 'DIRECT_DEBIT',
    BankTransfer: 'BANK_TRANSFER',
    Paylater: 'PAYLATER'
};
function InvoicePaymentMethodFromJSON(json) {
    return InvoicePaymentMethodFromJSONTyped(json, false);
}
exports.InvoicePaymentMethodFromJSON = InvoicePaymentMethodFromJSON;
function InvoicePaymentMethodFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.InvoicePaymentMethodFromJSONTyped = InvoicePaymentMethodFromJSONTyped;
function InvoicePaymentMethodToJSON(value) {
    return value;
}
exports.InvoicePaymentMethodToJSON = InvoicePaymentMethodToJSON;
