// xendit-node/invoice/models/NotificationChannel.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationChannelToJSON = exports.NotificationChannelFromJSONTyped = exports.NotificationChannelFromJSON = exports.NotificationChannel = void 0;
/**
 * Representing a notification channel for sending messages.
 * @export
 */
exports.NotificationChannel = {
    Email: 'email',
    Sms: 'sms',
    Whatsapp: 'whatsapp',
    Viber: 'viber'
};
function NotificationChannelFromJSON(json) {
    return NotificationChannelFromJSONTyped(json, false);
}
exports.NotificationChannelFromJSON = NotificationChannelFromJSON;
function NotificationChannelFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.NotificationChannelFromJSONTyped = NotificationChannelFromJSONTyped;
function NotificationChannelToJSON(value) {
    return value;
}
exports.NotificationChannelToJSON = NotificationChannelToJSON;
