// xendit-node/invoice/models/RetailOutletName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetailOutletNameToJSON = exports.RetailOutletNameFromJSONTyped = exports.RetailOutletNameFromJSON = exports.RetailOutletName = void 0;
/**
 * Representing the available retail outlet channels used for invoice-related transactions.
 * @export
 */
exports.RetailOutletName = {
    Alfamart: 'ALFAMART',
    Indomaret: 'INDOMARET',
    _7Eleven: '7ELEVEN',
    Cebuana: 'CEBUANA',
    DpEcpayLoan: 'DP_ECPAY_LOAN',
    DpMlhuillier: 'DP_MLHUILLIER',
    DpPalawan: 'DP_PALAWAN',
    DpEcpaySchool: 'DP_ECPAY_SCHOOL',
    Lbc: 'LBC'
};
function RetailOutletNameFromJSON(json) {
    return RetailOutletNameFromJSONTyped(json, false);
}
exports.RetailOutletNameFromJSON = RetailOutletNameFromJSON;
function RetailOutletNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.RetailOutletNameFromJSONTyped = RetailOutletNameFromJSONTyped;
function RetailOutletNameToJSON(value) {
    return value;
}
exports.RetailOutletNameToJSON = RetailOutletNameToJSON;
