// xendit-node/balance_and_transaction/models/TransactionTypes.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionTypesToJSON = exports.TransactionTypesFromJSONTyped = exports.TransactionTypesFromJSON = exports.TransactionTypes = void 0;
/**
 *
 * @export
 */
exports.TransactionTypes = {
    BatchDisbursement: 'BATCH_DISBURSEMENT',
    Disbursement: 'DISBURSEMENT',
    Payment: 'PAYMENT',
    Remittance: 'REMITTANCE',
    RemittancePayout: 'REMITTANCE_PAYOUT',
    RemittanceCollection: 'REMITTANCE_COLLECTION',
    Transfer: 'TRANSFER',
    PlatformFee: 'PLATFORM_FEE',
    Refund: 'REFUND',
    Cashback: 'CASHBACK',
    Topup: 'TOPUP',
    Withdrawal: 'WITHDRAWAL',
    Other: 'OTHER'
};
function TransactionTypesFromJSON(json) {
    return TransactionTypesFromJSONTyped(json, false);
}
exports.TransactionTypesFromJSON = TransactionTypesFromJSON;
function TransactionTypesFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.TransactionTypesFromJSONTyped = TransactionTypesFromJSONTyped;
function TransactionTypesToJSON(value) {
    return value;
}
exports.TransactionTypesToJSON = TransactionTypesToJSON;
